pkg/inject/sidecar_builder.go

```diff
@@ -79,6 +83,26 @@ func updateEnvMapForEnvoy(vars EnvoyTemplateVariables, env map[string]string, vn
 		// Specify a port value to override the default X-Ray daemon port: 2000
 		env["XRAY_DAEMON_PORT"] = strconv.Itoa(int(vars.XrayDaemonPort))
 
+		// Override the default sampling rate of 0.05 (5%) for AWS X-Ray tracer
+		// The value should be specified as a decimal between 0 and 1.00 (100%)
+		samplingRate, ok := env["XRAY_SAMPLING_RATE"]
+		if ok {
+			// `podAnnotations` contains the sampling rate and gets preference over helm configuration
+			// For now delete this value from env so that we can validate before adding again
+			delete(env, "XRAY_SAMPLING_RATE")
+		} else {
+			// `podAnnotations` doesn't contain the sampling rate so get value from helm configuration
+			samplingRate = vars.XraySamplingRate
+		}
+		fixedRate, err := strconv.ParseFloat(samplingRate, 32)
+		if err != nil || float64(0) > fixedRate || float64(1) < fixedRate {
+			// TODO: Return error if this value is not a decimal between 0 & 1.00
```

Comment:
Missed TODO ?

Reply:
Initial plan was to just ignore if user set a bad value and continue to use xray default (0.05). But changed this to return error. Now when user set a bad samplingRate then the deployment will fail and they have to check for the reason at `kubectl describe replicaset... ` to see what happened.

Thanks.